test/kmeans.jl

```diff
@@ -32,6 +32,21 @@ end
     end
 end
 
+@testset "k=1 and k=n corner cases" begin
+    x = [0.5 1 2; 1 0.5 0; 3 2 1]
+    km1 = kmeans(x,1)
+    @test km1.centers == reshape([7. /6,0.5, 2.0], (3,1))
+    @test km1.counts == fill(3, (1))
+    @test km1.assignments == fill(1, (3))
+    km3 = kmeans(x,3)
+    @test km3.centers == x
+    @test km3.counts == fill(1, (3))
+    @test km3.assignments == collect(1:3)
+    w = Vector{Int}([1,2,3])
```

Comment:
I'd test with something different than `Int`, because it's the default type when no weights are specified.
Also use something different than *.assignments*:
```suggestion
    w = [0.5, 2.0, 1.0]
```